src/wp-includes/class-wp-block.php

```diff
@@ -607,3 +571,129 @@ public function render( $options = array() ) {
 		return $block_content;
 	}
 }
+
+// phpcs:disable Generic.Files.OneObjectStructurePerFile.MultipleFound
+/**
+ * HTML API Processor subclass implementing experimental set_inner_html.
+ *
+ * DO NOT USE THIS CLASS. Internal usage only.
+ *
+ * @access private
+ */
+class PrivateProcessor extends WP_HTML_Processor {
+
+	/**
+	 * Set the inner HTML of the currrent node.
+	 *
+	 * @todo This method needs to check if the inner HTML can leak out of the current node.
+	 *
+	 * @param string $html The inner HTML to set.
+	 *
+	 * @return bool True if the inner HTML was set, false otherwise.
+	 */
+	public function set_inner_html( string $html ): bool {
+		if ( $this->is_virtual() ) {
+			return false;
+		}
+
+		if ( $this->get_token_type() !== '#tag' ) {
+			return false;
+		}
+
+		if ( $this->is_tag_closer() ) {
+			return false;
+		}
+
+		if ( ! $this->expects_closer() ) {
+			return false;
+		}
+
+		// @todo check if this is necessary
+		/*if (*/
+		/*  'html' !== $this->state->current_token->namespace &&*/
+		/*  $this->state->current_token->has_self_closing_flag*/
+		/*) {*/
+		/*  return false;*/
+		/*}*/
+
+		if ( '' !== $html ) {
+			$fragment_parser = $this->create_fragment_at_current_node( $html );
```

Comment:
I was testing the pull request, and it seems this method is private. If I change it to `protected`, everything seems to work as expected.

Reply:
Thank you! The changes (that I rebased this on) were partially reverted. This method may become public again in https://github.com/WordPress/wordpress-develop/pull/7912.

For now we can make it protected as part of this PR.